shared/src/main/java/com/couchbase/lite/Database.java

```diff
@@ -280,12 +284,15 @@ public void delete(Document document) throws CouchbaseLiteException {
     public void purge(Document document) throws CouchbaseLiteException {
         if (document == null)
             throw new IllegalArgumentException("a document parameter is null");
+        if (document.isNewDocument())
```

Comment:
check if revID is null

Reply:
Actually checking if revID is null is same as checking if c4doc is null. To be consistent, Document.isNewDocument() checks if getRevID() is null.